lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -391,12 +413,31 @@ defmodule Membrane.H264.FFmpeg.Parser do
     }
   end
 
-  defp carries_parameters_in_band?(<<payload::binary-size(40), _rest::binary>>) do
+  # Checks is the required parameter NALus (#{inspect}) are present in-band, before any video frames appears
+  defp carries_parameters_in_band?(payload) do
     types =
       NALu.parse(payload)
       |> elem(0)
       |> Enum.map(& &1.metadata.h264.type)
 
-    Enum.all?(@required_parameter_nalu, &Enum.member?(types, &1))
+    # split NALus parsed from the payload into two sections: parameters and data
+    # if data appears before required parameters, this would case an error in FFmpeg, so we evaluate the stream as not carrying parameters in-band
+    # if this is the case, they will be inserted into the stream before parsing, assuming that H264.RemoteStream caps are present
```

Comment:
```suggestion
    # Split NALus parsed from the payload into two sections: parameters and data.
    # If data appears before required parameters, this would cause an error in FFmpeg,
    # so we identify the stream as not carrying parameters in-band.
    # In such a case, they will be inserted into the stream before parsing,
    # assuming that H264.RemoteStream caps providing them are present
```